python/src/iceberg/utils/schema_conversion.py

```diff
@@ -0,0 +1,451 @@
+        Args:
+            field: The Avro field
+
+        Returns:
+            The Iceberg equivalent field
+        """
+        assert "field-id" in field, "Missing field-id in the Avro field, this is required for converting it to an Iceberg schema"
+
+        plain_type, is_optional = self._resolve_union(field["type"])
+
+        return NestedField(
+            field_id=field["field-id"],
+            name=field["name"],
+            field_type=self._convert_schema(plain_type),
+            is_optional=is_optional,
+            doc=field.get("doc"),
+        )
+
+    def _convert_record_type(self, record_type: Dict[str, Any]) -> StructType:
+        """
+        Converts the fields from a record into an Iceberg struct
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> record_type = {
+            ...     "type": "record",
+            ...     "name": "r508",
+            ...     "fields": [{
+            ...         "name": "contains_null",
+            ...         "type": "boolean",
+            ...         "doc": "True if any file has a null partition value",
+            ...         "field-id": 509,
+            ...      }, {
+            ...          "name": "contains_nan",
+            ...          "type": ["null", "boolean"],
+            ...          "doc": "True if any file has a nan partition value",
+            ...          "default": None,
+            ...          "field-id": 518,
+            ...      }],
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_record_type(record_type)
+            >>> expected = StructType(
+            ...     fields=(
+            ...         NestedField(
+            ...             field_id=509,
+            ...             name="contains_null",
+            ...             field_type=BooleanType(),
+            ...             is_optional=False,
+            ...             doc="True if any file has a null partition value",
+            ...         ),
+            ...         NestedField(
+            ...             field_id=518,
+            ...             name="contains_nan",
+            ...             field_type=BooleanType(),
+            ...             is_optional=True,
+            ...             doc="True if any file has a nan partition value",
+            ...         ),
+            ...     )
+            ... )
+            >>> expected == actual
+            True
+
+        Args:
+            record_type: The record type itself
+
+        Returns:
+        """
+        return StructType(*[self._convert_field(field) for field in record_type["fields"]])
+
+    def _convert_array_type(self, array_type: Dict[str, Any]) -> ListType:
+        element_type, element_is_optional = self._resolve_inner_type(array_type, "items", "element-id")
+        return ListType(element_id=array_type["element-id"], element_type=element_type, element_is_optional=element_is_optional)
```

Comment:
`_convert_field` uses an assertion to ensure that `field-id` is present. Should this do the same?

Reply:
Seems fair, I've added the check 👍🏻